hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -118,50 +119,52 @@ public void handle(OutOfMemoryError error) {
-        final ClientNetworkConfig networkConfig = config.getNetworkConfig();
+        this.loggingService = client.getLoggingService();
+        this.logger = loggingService.getLogger(ClientConnectionManager.class);
+
+        ClientConfig config = client.getClientConfig();
+        ClientNetworkConfig networkConfig = config.getNetworkConfig();
 
         final int connTimeout = networkConfig.getConnectionTimeout();
-        connectionTimeout = connTimeout == 0 ? Integer.MAX_VALUE : connTimeout;
+        this.connectionTimeout = connTimeout == 0 ? Integer.MAX_VALUE : connTimeout;
 
         HazelcastProperties hazelcastProperties = client.getProperties();
         long timeout = hazelcastProperties.getMillis(HEARTBEAT_TIMEOUT);
         this.heartbeatTimeout = timeout > 0 ? timeout : Integer.parseInt(HEARTBEAT_TIMEOUT.getDefaultValue());
 
         long interval = hazelcastProperties.getMillis(HEARTBEAT_INTERVAL);
-        heartbeatInterval = interval > 0 ? interval : Integer.parseInt(HEARTBEAT_INTERVAL.getDefaultValue());
+        this.heartbeatInterval = interval > 0 ? interval : Integer.parseInt(HEARTBEAT_INTERVAL.getDefaultValue());
 
-        executionService = (ClientExecutionServiceImpl) client.getClientExecutionService();
-        loggingService = client.getLoggingService();
+        this.executionService = (ClientExecutionServiceImpl) client.getClientExecutionService();
+        this.socketOptions = networkConfig.getSocketOptions();
 
         initializeSelectors(client);
 
-        socketOptions = networkConfig.getSocketOptions();
         ClientExtension clientExtension = client.getClientExtension();
-        socketChannelWrapperFactory = clientExtension.createSocketChannelWrapperFactory();
-        socketInterceptor = initSocketInterceptor(networkConfig.getSocketInterceptorConfig());
-        logger = loggingService.getLogger(ClientConnectionManager.class);
-        credentials = client.getCredentials();
+        this.socketChannelWrapperFactory = clientExtension.createSocketChannelWrapperFactory();
+        this.socketInterceptor = initSocketInterceptor(networkConfig.getSocketInterceptorConfig());
+
+        this.credentials = client.getCredentials();
     }
 
     protected void initializeSelectors(HazelcastClientInstanceImpl client) {
-        inputThread = new NonBlockingIOThread(
-                client.getThreadGroup(),
-                client.getName() + ".thread-in",
-                loggingService.getLogger(NonBlockingIOThread.class),
-                outOfMemoryHandler);
-        client.getMetricsRegistry().scanAndRegister(inputThread, "tcp." + inputThread.getName());
-
-        outputThread = new NonBlockingIOThread(
-                client.getThreadGroup(),
-                client.getName() + ".thread-out",
-                loggingService.getLogger(NonBlockingIOThread.class),
-                outOfMemoryHandler);
-        client.getMetricsRegistry().scanAndRegister(outputThread, "tcp." + outputThread.getName());
```

Comment:
I get your comment now. Yes. The IOThreadModel takes care of the metrics on the threads and whatever else is required.